TestImpactFramework is failing to build using VS2022

**Describe the bug**
The TestImpactFramework library is failing to build due to using a static_assert that is always false.
The compiler is allowed to trigger an error when a static_assert which doesn't depend on a template parameter is evaluated, even if a an instantiation of that code path within the template isn't taking.

The static assert is located in the [TestImpactClientSequenceReportSerializer.cpp](https://github.com/o3de/o3de/blob/development/Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/TestImpactClientSequenceReportSerializer.cpp#L753-L756) file

**Steps to reproduce**
Steps to reproduce the behavior:
1. Configure O3DE using cmake with VS2022 generator selected(This can be done via using the `cmake --preset windows-vs2022`) command
1. Open Visual Studio 2022 with generated project
1. Build the .vcxproj projects in the TestImpactFramework folder by right cliking on it and selecting the "Build" option
![image](https://user-images.githubusercontent.com/56135373/184222730-2a0c0f07-59d1-4c41-be2d-57b97b69e73d.png)
1. Watch the build fail with missing `typename` and `static_assert` errors

**Expected behavior**
The TestImpactFramework Visual Studio project builds successfully using VS2022

**Actual behavior**
The TestImpactFramework Visual Studio project fails using VS2022 with error
```
Build started...
1>------ Build started: Project: TestImpact.Runtime.Common.Static, Configuration: profile x64 ------
1>TestImpactClientSequenceReportSerializer.cpp
1>D:\work\o3de_dev\Code\Tools\TestImpactFramework\Runtime\Common\Code\Source\TestImpactClientSequenceReportSerializer.cpp(755,27): error C2338: static_assert failed: 'Template paramater must be a valid policy state type'
1>D:\work\o3de_dev\Code\Tools\TestImpactFramework\Runtime\Common\Code\Source\TestImpactClientSequenceReportSerializer.cpp(755,27): error C2338:             static_assert(false, "Template paramater must be a valid policy state type");
1>D:\work\o3de_dev\Code\Tools\TestImpactFramework\Runtime\Common\Code\Source\TestImpactClientSequenceReportSerializer.cpp(755,27): error C2338:                           ^
1>Done building project "TestImpact.Runtime.Common.Static.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - OS: [Windows]

--- a/Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/TestEngine/Common/TestImpactTestEngine.h
+++ b/Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/TestEngine/Common/TestImpactTestEngine.h
@@ -246,7 +246,7 @@ namespace TestImpact
         AZStd::optional<TestEngineJobCompleteCallback<TestTarget>> jobCallback,
         AZStd::optional<typename TestJobRunner::StdContentCallback> stdContentCallback)
     {
-        TestEngineJobMap<TestJobRunner::JobInfo::IdType, TestTarget> engineJobs;
+        TestEngineJobMap<typename TestJobRunner::JobInfo::IdType, TestTarget> engineJobs;
         auto [result, runnerJobs] = testRunner->RunTests(
             jobInfos,
             targetOutputCapture == Policy::TargetOutputCapture::None ? StdOutputRouting::None : StdOutputRouting::ToParent,

--- a/Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/TestImpactClientSequenceReportSerializer.cpp
+++ b/Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/TestImpactClientSequenceReportSerializer.cpp
@@ -752,7 +752,10 @@ namespace TestImpact
         }
         else
         {
-            static_assert(false, "Template paramater must be a valid policy state type");
+            // static assert needs to be depend on template parameters to defer evaluation until the function is instantiate
+            // with the type that isn't a valid policy state type, otherwise the compiler is free to evaluate the static assert
+            // without an instantiation of the template
+            static_assert(!AZStd::same_as<PolicyStateType, PolicyStateType>, "Template parameter must be a valid policy state type");
         }
     }
 
@@ -775,7 +778,7 @@ namespace TestImpact
             serialSequenceReportBase[SequenceReportFields::Keys[SequenceReportFields::MaxConcurrency]].GetUint64(),
             testTargetTimeout ? AZStd::optional<AZStd::chrono::milliseconds>{ testTargetTimeout } : AZStd::nullopt,
             globalTimeout ? AZStd::optional<AZStd::chrono::milliseconds>{ globalTimeout } : AZStd::nullopt,
-            DeserializePolicyStateType<SequenceReportBaseType::PolicyState>(serialSequenceReportBase),
+            DeserializePolicyStateType<typename SequenceReportBaseType::PolicyState>(serialSequenceReportBase),
             SuiteTypeFromString(serialSequenceReportBase[SequenceReportFields::Keys[SequenceReportFields::Suite]].GetString()),
             DeserializeTestSelection(serialSequenceReportBase[SequenceReportFields::Keys[SequenceReportFields::SelectedTestRuns]]),
             DeserializeTestRunReport(serialSequenceReportBase[SequenceReportFields::Keys[SequenceReportFields::SelectedTestRunReport]]));

